### ISSUE TITLE
Does this work on the Elite series 2?

### ISSUE THREAD

**Issue by arvz** [saqneo/Elite#7]
Elite series 2 just came out, and unfortunately it still has the same limitation in the paddles, not being able to be mapped as their own button

**Comment by saqneo** [comment]
Unfortunately it doesn't even work with the regular Elite anymore because
Microsoft removed the APIs I was using :(

**Comment by mooop12** [comment]
I haven't looked into it any closer yet, but there seems to be an api that allows reading of the paddles state: https://docs.microsoft.com/en-us/uwp/api/windows.gaming.input.gamepadbuttons

**Comment by saqneo** [comment]
Yeah, last time I took a look this API was hidden so I wasn't able to use
it effectively (rather not require copying files like I do with the current
implementation since so many people have had trouble with it). Maybe it's
possible now but I haven't tried to get it working on current versions of
Windows; I'll give updates if I ever take another shot.

**Comment by mooop12** [comment]
Quickly I ported the app to use the new api, but it does not seem to register the paddles, all other buttons are working. I tried an empty profile in the controller and I tried the default one, with the same effect. Any idea what this might be about?

**Comment by saqneo** [comment]
Iirc I wasn't able to instantiate the correct type of gamepad device
because the factory wasn't available to construct the one I thought would
support paddles (visible in the dll via reflection).

If you're a strong coder you could try instantiating the object in C++ via
COM, which I believe is what the app uses to communicate with the
controller since it existed long before the new APIs. Or maybe they've
changed it; if you're familiar with ildasm you can use reflection to find
out.

I'm home for the holidays right now so I can't take a look directly,
hopefully this is informative though.

**Comment by mooop12** [comment]
C++ is beyond my area of expertise I'm afraid. I'm not even sure which library implements these methods, the aforementioned namespace only contains externs for those methods and I'm not sure how to check which library it calls.

I also tried RawGameController (it would make sense for it to return raw, unmapped data, right?), but the effect is the same, paddles are mapped to the buttons from the Xbox Accessories app.

I hope you can get it working when you're back, good luck. In the meantime, Merry Christmas.
